model-optimizer/extensions/ops/loop.py

```diff
@@ -312,14 +313,52 @@ def add_back_edge(loop_node: Node, internal_parameter: Node, internal_result: No
                                      'from_port': 0,
                                      'to_port': 0})
 
+    @staticmethod
+    def parameter_unchanged_after_iteration(loop_node: Node, body_parameter: Node):
+        """
+        Checks if the body Parameter node is connected to some body Result and the data provided to Result is not
+        changed between iterations. The data is considered unchanged if:
+        1. There is no back edge for this Parameter OR
+        2. There is a back edge from some Result to Parameter and there are only Identity ops in between or
+           Parameter is connected to Result directly.
+
+        :param loop_node: the Loop node to check
+        :param body_parameter: the body Parameter node
+        :return: the result of the check
+        """
+        assert body_parameter.id in loop_node.body
+        assert body_parameter.soft_get('op') == 'Parameter'
+        if not any([attr['to_layer'] == body_parameter.soft_get('internal_layer_id') for attr in loop_node.back_edges]):
+            return True
```

Comment:
nit: we know that the number of back edges is equal either to 0 or 1:
needed_back_edges = [attr['to_layer'] == body_parameter.soft_get('internal_layer_id') for attr in loop_node.back_edges]
if len(needed_back_edges) == 0:
    return True
assert len(needed_back_edges) == 1

back_edge_attrs = needed_back_edges[0]
result_internal_id = back_edge_attrs['from_layer']
result_nodes = loop_node.body.get_op_nodes(internal_layer_id=result_internal_id)
assert len(result_nodes) == 1, 'There should be exactly one node with id {}, but there are {}' \
                               ''.format(result_internal_id, len(result_nodes))
result_node = result_nodes[0]
# check that the Result node consumes data from Parameter node directly or through Identity operations
parameters = common_bfs(result_node, ['Identity'], ['Parameter'], is_backward=True, attr_to_check='op',
                        follow_multi_consumer_data_nodes=True)
if any([node.soft_get('internal_layer_id') == body_parameter.internal_layer_id for node in parameters]):
    return True

And you do not need `for` loop and `if` check then